### ISSUE TITLE
Revisiting standardize

### ISSUE THREAD

**Issue by RaulPPelaez** [torchmd/torchmd-net#277]
When enabled, the standardize functionality processes the whole Dataset and stores the mean and std of the energies (well, the "y" field) in it:

https://github.com/torchmd/torchmd-net/blob/6d8e3159cfb8bb971ecf7a2abd589735d79a7e53/torchmdnet/data.py#L172-L202

These are then used during prediction:

https://github.com/torchmd/torchmd-net/blob/6d8e3159cfb8bb971ecf7a2abd589735d79a7e53/torchmdnet/models/model.py#L358-L375

There are some issues we should consider with its current state:

1. There is no way to get unit std without shifting energies.
2. It has a potentially unwanted interaction with atomref.
3. For unit std it might be better to do batch normalization (which does not require going over the whole dataset before training)

I am opening this issue to start a discussion on what to do about these.

**Comment by guillemsimeon** [comment]
I can say something here. Standardize was only used in (r)MD17 (and it didn't even make sense). Since MD17 consists in training on just one system, people were using at some point the standard deviation and the mean to make learning easier. Of course this does not apply when building general NNPs, and one can argue that it does not even make sense in the single system case. To the best of my knowledge, it has never been used apart from (r)MD17. The way to go with NNPs, imo, is by means of atomic reference energies (even if they are learnable), because this can scale to arbitrary systems. Since you are also currently having the discussion about Atomref, my point would be that having Atomref-like behavior inside the full model makes infinitely more sense than having standardize.

**Comment by guillemsimeon** [comment]
I cannot come up with any use of standardize apart from rMD17, and the
abstract case of scaling the energies to std one. In the reality you will
never have a std to refer to when doing inference, it is an artifact of the
training set. Allegro’s paper has a nice explanation on target
normalizations.

**Comment by RaulPPelaez** [comment]
Thanks for the wonderful insights guys.

It is becoming clear to me that we should:
1. Build the current Atomref prior into de model (at the same conceptual level as standardize is now)
2. Transform the current Atomref prior into something like "LearnablePerAtomEnergyOffset"
3. Separate unit-std from energy displacement in standardize
4. Allow customization of standardize energy displacement (use ref_energies, compute mean energy per element in the dataset...)

**Comment by guillemsimeon** [comment]
for me it is the most elegant and general way to proceed

**Comment by RaulPPelaez** [comment]
The documentation also requires the addition of a section like "Dataset standardisation". I will add it in the PR for this, will defo need yo to pour your knowledge into it.

**Comment by guillemsimeon** [comment]
I will be happy to help
